### PR TITLE
Move as many tests from tests/ui/numbers-arithmetic to tests/ui/lint as possible

### LINKED ISSUE #119851: Review tests in tests/ui/numbers-arithmetic for duplicates
https://github.com/rust-lang/rust/pull/119432 adds extensive tests for our overflow lint; as far as I can see they subsume most of what we have in tests/ui/numbers-arithmetic, so the old tests can be deleted. Yay for having the tests nicely grouped in one file rather than spread across a dozen files!

So, once #119432 lands, we should do a review of these tests and remove duplicates.

Cc @gurry

### PR DESCRIPTION
**Pull request by mj10021**
Fixes #119851 , and also consolidates as many individual tests as possible from numbers-arithmetic.  I might have moved the tests in too aggressively, so let me know

### CODE REVIEW

File: tests/ui/numbers-arithmetic/overflowing-neg.rs [hunk 1]
```diff
@@ -1,10 +0,0 @@
-//@ run-fail
-//@ error-pattern:attempt to negate with overflow
-//@ ignore-emscripten no processes
-//@ compile-flags: -C debug-assertions
-
-#![allow(arithmetic_overflow)]
-
-fn main() {
-    let _x = -i8::MIN;
```
Comment (nnethercote) [inline tests/ui/numbers-arithmetic/overflowing-neg.rs:9]: I think this one was missed?

Comment (nnethercote) [reply tests/ui/numbers-arithmetic/overflowing-neg.rs:9]: Otherwise, everything else looks ok.

Comment (mj10021) [reply tests/ui/numbers-arithmetic/overflowing-neg.rs:9]: isn't that one on line 225/226?

Comment (nnethercote) [reply tests/ui/numbers-arithmetic/overflowing-neg.rs:9]: True! Not sure how I overlooked that.

**Pull request merged**